presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/LongDecimalWithOverflowStateSerializer.java

```diff
@@ -33,21 +34,33 @@ public Type getSerializedType()
     }
 
     @Override
-    public void serialize(BigIntegerState state, BlockBuilder out)
+    public void serialize(LongDecimalWithOverflowState state, BlockBuilder out)
     {
-        if (state.getBigInteger() == null) {
+        if (state.getLongDecimal() == null) {
             out.appendNull();
         }
         else {
-            VARBINARY.writeSlice(out, Slices.wrappedBuffer(state.getBigInteger().toByteArray()));
+            byte[] longDecimalBytes = state.getLongDecimal().getBytes();
+
+            Slice slice = Slices.allocate(Long.BYTES + longDecimalBytes.length);
+            SliceOutput output = slice.getOutput();
+
+            output.writeLong(state.getOverflow());
```

Comment:
you could just use `VARBINARY.writeSlice(out, Slices.wrappedLongArray(state.getOverflow())`